backend/tilavarauspalvelu/api/graphql/types/application_section/types.py

```diff
@@ -115,3 +171,51 @@ def optimize_has_reservations(queryset: models.QuerySet, optimizer: QueryOptimiz
         )
 
         return queryset
+
+    def resolve_pindora_info(root: ApplicationSection, info: GQLInfo) -> PindoraSectionInfoData | None:
+        # Not using access codes
+        if not root.should_have_active_access_code:
+            return None
+
+        # No need to show Pindora info after 24 hours have passed since the section has ended
+        today = local_date()
+        cutoff = root.reservations_end_date + datetime.timedelta(hours=24)
+        if today > cutoff:
+            return None
+
+        has_perms = info.context.user.permissions.can_manage_application(root.application, reserver_needs_role=True)
+
+        # Don't show Pindora info without permissions if the application round results haven't been sent yet
+        if not has_perms and root.application.application_round.sent_date is None:
+            return None
+
+        try:
+            response = PindoraClient.get_seasonal_booking(section=root.ext_uuid)
+        except Exception:  # noqa: BLE001
+            return None
+
+        # Don't show Pindora info without permissions if the access code is not active
+        access_code_is_active = response["access_code_is_active"]
+        if not has_perms and not access_code_is_active:
+            return None
+
+        return PindoraSectionInfoData(
+            access_code=response["access_code"],
+            access_code_generated_at=response["access_code_generated_at"],
+            access_code_is_active=response["access_code_is_active"],
+            access_code_keypad_url=response["access_code_keypad_url"],
+            access_code_phone_number=response["access_code_phone_number"],
+            access_code_sms_number=response["access_code_sms_number"],
+            access_code_sms_message=response["access_code_sms_message"],
+            access_code_validity=[
+                PindoraSectionValidityInfoData(
```

Comment:
On the app.section level the `reservation_unit_id` should also be here... but the uuid is not that practical for the UI.
I guess it could be mapped to the pk since the section knows which reservation units it has allocations (and only those need to be pre-fetched).

Reply:
Added reservation and series ids to the validity info to uniquely identify them. Requires 2N + 1 queries to resolve them for application sections, but it should be OK if used for a single section.